File: session-02-dependency-injection/java/src/main/java/coffeeshop/menu/MenuService.java
package coffeeshop.menu;

import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class MenuService {

    private final MenuRepository repository;

    public MenuService(MenuRepository repository) {
        this.repository = repository;
    }

    public List<MenuItem> listItems() {
        return repository.findAll();
    }
}

File: session-04-proxy-and-aop/typescript/src/menu/menu.service.ts
import {Injectable} from "@nestjs/common";
import {MenuItem} from "./menu.model";
import {MenuRepository} from "./menu.repository";

// Clean service: interceptors handle cross-cutting concerns.
@Injectable()
export class MenuService {
    constructor(private readonly repository: MenuRepository) {
    }

    listItems(): MenuItem[] {
        return this.repository.findAll();
    }
}
